packages/flutter/lib/src/material/text_field.dart

```diff
@@ -1000,7 +1000,7 @@ class _TextFieldState extends State<TextField> with RestorationMixin implements
 
   bool get _hasError => widget.decoration?.errorText != null || widget.decoration?.error != null || _hasIntrinsicError;
 
-  Color get _errorColor => widget.cursorErrorColor ?? widget.decoration?.errorStyle?.color ?? Theme.of(context).colorScheme.error;
+  Color get _errorColor => widget.cursorErrorColor ?? _getEffectiveDecoration().errorStyle?.color ?? Theme.of(context).colorScheme.error;
```

Comment:
I wonder if there's anywhere else that should be using _getEffectiveDecoration instead of widget.decoration?

Reply:
I checked this out and didn't see anything else that stood out. Other uses like accessing `enabled`, `counterText`, `errorText` or `hintText` are okay because those are not inherited from an `InputDecorationTheme`.